default/scripting/ship_parts/Weapons.focs.txt

```diff
@@ -147,3 +157,18 @@ WEAPON_BASE_UNOWNED_EFFECTS
                 SetMaxSecondaryStat partname = "@1@" value = Value + PartSecondaryStat name = "@1@"
             ]
 '''
+
+WEAPON_BASE_LOCKED_EFFECTS
+'''        EffectsGroup
+            scope = Source
+            activation = And [
+                Not Unowned
+                Not OwnerHasShipPartAvailable name = "@1@"
+            ]
+            stackinggroup = "WEAPON_BASE_LOCKED_EFFECTS_@1@"
+            accountinglabel = "@1@"
+            effects = [
+                SetMaxCapacity partname = "@1@" value = Value + PartCapacity name = "@1@"
+                SetMaxSecondaryStat partname = "@1@" value = Value + PartSecondaryStat name = "@1@"
+            ]
+'''
```

Comment:
The effect is now the same as the unowned macro. Seems like merging them into one macro (activation = Unowned or Not OwnerHasShipPartAvailable) would make more sense.